nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/hint/MoreLikeThisESHintQueryBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.elasticsearch.hint;
+
+import static org.nuxeo.elasticsearch.ElasticSearchConstants.DOC_TYPE;
+
+import org.elasticsearch.index.query.MoreLikeThisQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.query.sql.model.EsHint;
+import org.nuxeo.elasticsearch.api.ESHintQueryBuilder;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * The implementation of {@link ESHintQueryBuilder} for the <strong>"more_like_this"</strong> ES hint operator.
+ *
+ * @since 11.1
+ */
+public class MoreLikeThisESHintQueryBuilder implements ESHintQueryBuilder {
+
+    public static final int MORE_LIKE_THIS_MIN_TERM_FREQ = 1;
+
+    public static final int MORE_LIKE_THIS_MIN_DOC_FREQ = 3;
+
+    public static final int MORE_LIKE_THIS_MAX_QUERY_TERMS = 12;
+
+    /**
+     * @return {@link org.elasticsearch.index.query.MoreLikeThisQueryBuilder}
+     */
+    @Override
+    public QueryBuilder make(EsHint hint, String fieldName, Object value) {
+        String[] indexFields = hint.getIndexFieldNames();
+        String[] fields = indexFields.length > 0 ? indexFields : new String[] { fieldName };
+        return QueryBuilders.moreLikeThisQuery(fields, null, getItems(value))
+                            .analyzer(hint.analyzer)
+                            .minDocFreq(MORE_LIKE_THIS_MIN_DOC_FREQ)
```

Comment:
`QueryBuilders` often have additional parameters.  It would be good if the `ESHintQueryBuilder` descriptor had a mechanism to pass in parameters to the implentation.  For example, take a look at `CodecDescriptor`.

Reply:
That way, all the `MORE_LIKE_THIS_` params could be defaults rather than hard-coded.

Reply:
It looks like to query parameters rather than configuration ones to me. In a same way than `relation` in `geo_shape` query. A solution could be to improve `EsHint` to hold these extra parameters. Maybe not needed at the moment (?)

Reply:
To me, this PR is a combination of query and configuration parameters.  I like it though!  I am advocating flexible `options`.  That would enable multiple configs, eg. morelike1, morelike2, morelike3 via extension points.  There doesn't appear to be a mechanism to pass query parameters for this?  Perhaps I am missing something?

Reply:
I see your point, indeed it would be nice to have them. Probably in another ticket? @troger, @RSalem07 wdyt?

Reply:
yes and If you want to use advanced options just contribute a new hint